lib/elixir/test/elixir/calendar/duration_test.exs

```diff
@@ -8,21 +8,21 @@ defmodule DurationTest do
     assert Duration.new!(year: 2, month: 1, week: 3) == %Duration{year: 2, month: 1, week: 3}
     assert Duration.new!(microsecond: {20000, 2}) == %Duration{microsecond: {20000, 2}}
 
-    assert_raise KeyError, ~s/key :months not found/, fn ->
-      Duration.new!(months: 1)
-    end
+    assert_raise ArgumentError,
+                 "expected an integer for :month, got nil",
+                 fn -> Duration.new!(month: nil) end
+
+    assert_raise ArgumentError,
+                 "unexpected unit :years",
+                 fn -> Duration.new!(years: 1) end
```

Comment:
I genuinely made that mistake and had to stop and think for a second - almost wondered if we should add a "Did you meant `:year`?"

Reply:
that'd be icing on the cake of course, since we figured plural units are widely used, it might be worth adding

Reply:
In another PR if we want to, plz 🙃